File: cp_rust/src/bin/abc409-b-citation.rs
// Created by Ayush Biswas at 2026/07/08 13:45
// https://atcoder.jp/contests/adt_medium_20250723_3/tasks/abc409_b
use cp_lib::*;

// @code begin
use cpio::*;

sol! {
    fn solution(
        n: usize,
        (mut a): [usize]
    ) -> usize {
        a.sort();
        let mut res = 0;
        for (i, ai) in a.into_iter().enumerate() {
            res = res.max(ai.min(n - i));
        }
        res
    }
}
// @code end

File: cp_nim/src/bin/abc409-b-citation.nim
# Created by Ayush Biswas at 2026/07/19 08:35
# https://atcoder.jp/contests/adt_medium_20250723_3/tasks/abc409_b
import ../lib/cpio

# @code begin

import algorithm, math

var r = newReader()

let
  n = r[int]
  a = r[seq[int]].sorted

var res: int
for i, x in a.pairs:
  let el = (n - i)
  res = res.max (x.min el)

echo res
# @code end
